### DISCUSSION TITLE
Raise an (Alert) Event for a group of sensors

### DISCUSSION THREAD

**Discussion by kt-alt** [nicolargo/glances#2692] [Q&A] [answered]
A few days ago, my server unexpectedly shutdown due to high temp.  I've installed glances monitor it and can see the temp readings, but it doesn't appear that an alert or event is raised for temperature.  

I have previously modified event.py to create syslog messages for events, which are sent over to graylog for centralized alerting.  I'd like to do this for my server temp as well.

Is there a way to enable this? Or is this something to suggest as a new feature.

**Comment by nicolargo** [comment]
Hi @kt-alt 

yes, you can configure thresholds for the sensors in the Glances configuration file:

```
[sensors]
# Documentation: https://glances.readthedocs.io/en/latest/aoa/sensors.html
disable=False
# By default refresh every refresh time * 2
#refresh=6
# Hide some sensors
#hide=ambient
# Sensors core thresholds (in Celsius...)
# Default values are grabbed from the system
temperature_core_careful=60
temperature_core_warning=70
temperature_core_critical=80
...
```

**Reply by kt-alt** [reply]
Thanks, I misunderstood the note at the bottom of this part of the docs - which noted no alerts on sensors.

https://glances.readthedocs.io/en/latest/aoa/sensors.html

I did try to stress the system to trigger a temperature warning but was unsuccessful.  Perhaps I didn't exceed the default values from the system, so will try setting the custom values.

**Reply by kt-alt** [reply]
@nicolargo - I set some purposefully low custom thresholds in the conf file and was able to see the the temperature readings in the UI were red (meaning the thresholds were adjusted).  However they did not trigger any events either in the UI or in the syslog.

Is this something possible or could be done with small code adjustments?  It wasn't obvious from the event.py what customizations might be needed and I'd rather not break stuff blindly.

Thanks.

**Reply by vlanx** [reply]
Did you manage to pull this off? I'm interested in the same setup, but even changing the threshold values to some low ones, doesn't change the color output in the dashboard.

<img width="1710" alt="Screenshot 2024-12-18 at 23 35 25" src="https://github.com/user-attachments/assets/bb5bee61-e563-47af-ac07-8f6c4a5deb9b" />

@nicolargo Do you have any idea why the threshold values are not being taken into account?

**Reply by kt-alt** [reply]
Not fully, I eventually gave up and switched to prometheus as I think I was trying to get glances do stuff too outside of what it was intended to do.

**Comment by nicolargo** [comment]
I look deeper in the code and the current behavor is the following:
- search  limit define for a **specific** sensor in the Glances configuration file (ex: temperature_core_Core 4_careful=10, temperature_core_Core 4_warning=20,temperature_core_Core 4_critical=30)
- if the following limit is not define, the uses system one

Perhaps a enhancement will be:
- search  limit define for a **specific** sensor in the Glances configuration file (ex: temperature_core_Core 4_careful=10, temperature_core_Core 4_warning=20,temperature_core_Core 4_critical=30)
- _if the previous limit is not define, search  limit define for a **type** sensor in the Glances configuration file (ex: temperature_core_careful=10, temperature_core_warning=20,temperature_core_critical=30)_
- if the following limit is not define, the uses system one

What do you think ?

**Reply by vlanx** [reply]
Hi @nicolargo,

I do agree that the enhancement you proposed would give even more granularity in defining the warning and critical values for the sensors, which, in my opinion, is a good thing. For example, I would be interested in defining the temperatures for all cores at once, I would then use `temperature_core_warning=40`.

But even now, defining the various temperature limits is still not possible for me. Glances is not taking them into account.
<img width="1710" alt="Screenshot 2024-12-19 at 18 38 24" src="https://github.com/user-attachments/assets/8e117d6e-c142-422d-978a-bc61ee822c00" />

```
# glances --version
Glances version:        4.2.1
Glances API version:    4
PsUtil version:         6.1.0
Log file:               /root/.local/share/glances/glances.log
```

**Comment by nicolargo** [comment]
Merged into develop (see https://github.com/nicolargo/glances/issues/3049)

@kt-alt @dot-1q can you test the feature ?

**Reply by vlanx** [reply, marked as answer]
Hi @nicolargo,
Thank you for your quickness in viewing this issue and providing help and a solution!

Today I tested your `dev` branch locally and the colors are now being outputted correctly on my machine.
I can set colors both globally for the temperature sensor, and also for a specific one (Core 0). I tried to showcase that in the screenshot.

<img width="1710" alt="Screenshot 2024-12-22 at 13 51 23" src="https://github.com/user-attachments/assets/fd302549-c5fa-4573-91f2-1fd69f0cda2b" />

It is working as I would've expected now. Thank you so much again!
